Hoxtygen/home360-server
# Feat: implement access and refresh token

## What does this PR do?
Implements access and refresh tokens

## Description of the task to be completed
* Create TokenResponse and TokenRequest DTO's
* Create the different layers and implementation for refresh token
* incorporate token into authentication services

## How should this be manually tested?
* Clone the repository.
* On your PC open a terminal and run git clone <repo-url>.
* Open the pom.xml file and download dependencies.
* Run the application.
* Open Postman, Insomnia or any HTTP client.
* Register as a new user by sending a `POST` request to `http://localhost:8080/api/v1/auth/register`
```
curl --location 'http://localhost:8080/api/v1/auth/register' \
--header 'Content-Type: application/json' \
--data-raw '{
   "firstName": "Mimosa",
    "lastName": "Pudica",
    "email": "user@example.com",
    "phoneNumber": "09012345678",
    "address": "221B, Baker street London",
    "password": "Int3rnat!onalization"
}'
```
* Check you email and click on the verification link sent to verify your account
* Login to the application by sending a `POST` request to `http://localhost:8080/api/v1/auth/login` with your credentials
```
curl --location 'http://localhost:8080/api/v1/auth/login' \
--header 'Content-Type: application/json' \
--data-raw '{
   "email": "user@example.com",
    "password": "Int3rnat!onalization"
}
'

```
* To fetch a new access token send a `POST` request to `http://localhost:8080/api/v1/auth/refreshToken` passing the `refresh token` to the. request body
```
curl --location 'http://localhost:8080/api/v1/auth/refreshToken' \
--header 'Content-Type: application/json' \
--data '{
    "token": "453f43de-baba-4244-b383-48ab73d2dbc1"
}'
```


## What are the relevant Jira board stories?
[HOM-48](https://wasiu-dev.atlassian.net/jira/software/projects/HOM/boards/2?selectedIssue=HOM-48)

## Screenshots (if appropriate)
 | login request|new access token request|
|---|---|
|<img width="1680" alt="Screenshot 2023-11-20 at 21 34 29" src="https://github.com/Hoxtygen/home360-server/assets/19408669/61fb5976-5375-400b-9702-6f7737c2d5c2"> |<img width="1680" alt="Screenshot 2023-11-20 at 21 35 50" src="https://github.com/Hoxtygen/home360-server/assets/19408669/053eeeae-e146-483b-bd7d-ef45f14b2182"> |

[HOM-48]: https://wasiu-dev.atlassian.net/browse/HOM-48?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
